modules/EvseV2G/connection/connection.cpp

```diff
@@ -192,6 +192,10 @@ int check_interface(struct v2g_context* v2g_ctx) {
         v2g_ctx->if_name = choose_first_ipv6_interface();
     }
 
+    if (v2g_ctx->if_name == nullptr) {
```

Comment:
In line 205 there is already a check if `if_name == nulltpr`. So the segfault is happening in line 199 (`mreq.ipv6mr_interface = if_nametoindex(v2g_ctx->if_name);`) ?

Reply:
Just tried a simple test program and strcmp doesn't like nullptr as an argument.
There could be a segfault at line 191

Reply:
Yes the segfault happened in `if_nametoindex()` after `choose_first_ipv6_interface();` returned nullptr. I agree that we should already return early if `v2g_ctx->if_name` is already nullptr in the `strcmp` in line 191

Reply:
`v2g_ctx->if_name` can only be a `nullptr` if `choose_first_ipv6_interface()` returns nullptr. Before that it is a string from `config.device`. So strcmp should not be a problem.

Reply:
I've added some additional checks at the beginning of the function